File: karton-backend/app/Models/Ingredient.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;

class Ingredient extends Model
{
    use HasFactory;

    protected $fillable = [
        'name',
        'unit',
    ];
    
    public function products()
    {
        return $this->belongsToMany(
            Product::class,
            'product_ingredient',
            'ingredient_id',  
            'product_id'      
        )->withPivot('quantity');
    }

    public function inventory()
    {
        return $this->hasOne(Inventory::class);
    }

    public function stockMovements()
    {
        return $this->hasMany(StockMovement::class);
    }
}

File: LeltarSupportMauiApp/Models/Ingredient.cs
using System.Collections.Generic;

namespace LeltarSupportMauiApp.Models
{
    public class Ingredient
    {
        public int Id { get; set; }
        public string Name { get; set; } = null!;
        public string Unit { get; set; } = null!;

        // Navigation
        public IList<ProductIngredient> ProductIngredients { get; set; } = new List<ProductIngredient>();
        public Inventory? Inventory { get; set; }
        public IList<StockMovement> StockMovements { get; set; } = new List<StockMovement>();
        public IList<PurchaseOrderItem> PurchaseOrderItems { get; set; } = new List<PurchaseOrderItem>();
    }
}